src/index.ts

```diff
@@ -267,38 +334,25 @@ class Logging {
       throw new Error('A sink configuration object must be provided.');
     }
     if (common.util.isCustomType(config.destination, 'bigquery/dataset')) {
-      this.setAclForDataset_(name, config, callback);
-      return;
+      await this.setAclForDataset_(config);
     }
     if (common.util.isCustomType(config.destination, 'pubsub/topic')) {
-      this.setAclForTopic_(name, config, callback);
-      return;
+      await this.setAclForTopic_(config);
     }
     if (common.util.isCustomType(config.destination, 'storage/bucket')) {
-      this.setAclForBucket_(name, config, callback);
-      return;
+      await this.setAclForBucket_(config);
     }
     const reqOpts = {
       parent: 'projects/' + this.projectId,
       sink: extend({}, config, {name}),
     };
     delete reqOpts.sink.gaxOptions;
-    this.request(
-        {
-          client: 'ConfigServiceV2Client',
-          method: 'createSink',
-          reqOpts,
-          gaxOpts: config.gaxOptions,
-        },
-        (err, resp) => {
-          if (err) {
-            callback(err, null, resp);
-            return;
-          }
-          const sink = self.sink(resp.name);
-          sink.metadata = resp;
-          callback(null, sink, resp);
-        });
+    await this.setProjectId(reqOpts);
```

Comment:
this looks like a behavior change.  Why aren't we just calling `request` like previously?

Reply:
The `request` differentiates whether to make `makeRequestCallback` or `makeRequestStream` depending on whether `callback` is provided or not. 
By converting everything to async-first `callback` serves a flag for action to `callbackify`. As such I am proposing a different approach of methods to call `gax` client directly since `gax` client is available and is not loaded at runtime anymore. 

As I mentioned in the comment below, I will split this PR into multiple smaller PRs.